src/main/java/frc/robot/hardware/phoenix6/motor/TalonFXMotor.java

```diff
@@ -0,0 +1,96 @@
+package frc.robot.hardware.phoenix6.motor;
+
+import com.ctre.phoenix6.configs.TalonFXConfiguration;
+import com.ctre.phoenix6.signals.NeutralModeValue;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.wpilibj2.command.sysid.SysIdRoutine;
+import frc.robot.Robot;
+import frc.robot.hardware.interfaces.ControllableMotor;
+import frc.robot.hardware.phoenix6.Phoenix6Device;
+import frc.robot.hardware.interfaces.IRequest;
+import frc.robot.hardware.phoenix6.Phoenix6DeviceID;
+import frc.robot.hardware.phoenix6.motor.simulation.TalonFXSimulation;
+import frc.robot.hardware.mechanisms.MechanismSimulation;
+import frc.robot.hardware.phoenix6.request.Phoenix6Request;
+import frc.utils.alerts.Alert;
+import frc.utils.calibration.sysid.SysIdCalibrator;
+
+import java.util.Optional;
+
+public class TalonFXMotor extends Phoenix6Device implements ControllableMotor {
+
+	private static final int APPLY_CONFIG_RETRIES = 5;
+
+	private final TalonFXWrapper motor;
+	private final Optional<TalonFXSimulation> talonFXSimulationOptional;
+	private final SysIdCalibrator.SysIdConfigInfo sysidConfigInfo;
+
+	public TalonFXMotor(
+		String logPath,
+		Phoenix6DeviceID deviceID,
+		TalonFXConfiguration configuration,
+		SysIdRoutine.Config sysidConfig,
+		MechanismSimulation simulation
+	) {
+		super(logPath);
+		this.motor = Robot.ROBOT_TYPE.isSimulation() ? new TalonFXWrapper(deviceID.ID()) : new TalonFXWrapper(deviceID);
+		if (!motor.applyConfiguration(configuration, APPLY_CONFIG_RETRIES).isOK()) {
+			new Alert(Alert.AlertType.ERROR, getLogPath() + "ConfigurationFailed").report();
+		}
+		this.talonFXSimulationOptional = Robot.ROBOT_TYPE.isSimulation() && simulation != null
+			? Optional.of(new TalonFXSimulation(motor, configuration, simulation))
+			: Optional.empty();
+		this.sysidConfigInfo = new SysIdCalibrator.SysIdConfigInfo(sysidConfig, true);
+		motor.optimizeBusUtilization();
```

Comment:
This is too much logic for a constructor, refactor to use functions

Reply:
in pr